sentry-spring/src/main/java/io/sentry/spring/reactive/SentryReactiveWebRequestProcessor.java

```diff
@@ -0,0 +1,68 @@
+package io.sentry.spring.reactive;
+
+import com.jakewharton.nopen.annotation.Open;
+import io.sentry.EventProcessor;
+import io.sentry.SentryEvent;
+import io.sentry.SentryOptions;
+import io.sentry.protocol.Request;
+import io.sentry.util.Objects;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.server.reactive.ServerHttpRequest;
+
+/** Attaches information about Reactive HTTP request to {@link SentryEvent}. */
+@Open
+public class SentryReactiveWebRequestProcessor implements EventProcessor {
+  private static final List<String> SENSITIVE_HEADERS =
+      Arrays.asList("X-FORWARDED-FOR", "AUTHORIZATION", "COOKIE");
+
+  private final @NotNull ServerHttpRequest request;
+  private final @NotNull SentryOptions options;
+
+  public SentryReactiveWebRequestProcessor(
+      final @NotNull ServerHttpRequest request, final @NotNull SentryOptions options) {
+    this.request = Objects.requireNonNull(request, "request is required");
+    this.options = Objects.requireNonNull(options, "options are required");
+  }
+
+  @Override
+  public @NotNull SentryEvent process(
+      final @NotNull SentryEvent event, final @Nullable Object hint) {
+    event.setRequest(resolveSentryRequest(request));
+    return event;
+  }
+
+  private @NotNull Request resolveSentryRequest(final @NotNull ServerHttpRequest httpRequest) {
+    final Request sentryRequest = new Request();
+    sentryRequest.setMethod(httpRequest.getMethodValue());
+    sentryRequest.setQueryString(httpRequest.getURI().getQuery());
+    sentryRequest.setUrl(httpRequest.getURI().toString().replaceFirst("\\?.*$", ""));
+    sentryRequest.setHeaders(resolveHeadersMap(httpRequest));
+    if (options.isSendDefaultPii()) {
+      sentryRequest.setCookies(
+          toString(httpRequest.getHeaders().getValuesAsList(HttpHeaders.COOKIE)));
+    }
+    return sentryRequest;
+  }
+
+  private @NotNull Map<String, String> resolveHeadersMap(
```

Comment:
There seems to be quite a bit copied over from: https://github.com/getsentry/sentry-java/blob/cd45528388f60b9f12b77e3cb59163be1371d693/sentry-spring/src/main/java/io/sentry/spring/SentryRequestHttpServletRequestProcessor.java

I'm afraid they might diverge in the future as we add stuff or fix bugs. Is it possible to pull this into a single implementation by sharing an abstraction?

Reply:
Of course it looks very similar, but in fact almost every line of code is different because it uses a different web api for access the request information, but I'll try to figure out a way to abstract that.

Reply:
If it's too much hassle, maybe it isnt' worth it.
A changelog entry would be great though